linkml/generators/pydanticgen.py

```diff
@@ -525,6 +532,52 @@ def generate_collection_key(
             return list(collection_keys)[0]
         return None
 
+    def get_array_representations_range(self, slot: SlotDefinition) -> str:
+        """
+        Generate the python range for array representations
+        """
+        array_info = slot.array
+        if array_info is None:  # pragma: no cover
+            return ""  # this should not happen
```

Comment:
should we raise `ValueError` here just to avoid silent bugs?

Reply:
oh wait wasn't this supposed to be "any shaped array"?

Reply:
Yep!